flink-end-to-end-tests/test-scripts/test_ha_dataset.sh

```diff
@@ -53,20 +50,51 @@ function run_ha_test() {
 
     wait_job_running ${JOB_ID}
 
-    # start the watchdog that keeps the number of JMs stable
-    start_ha_jm_watchdog 1 "StandaloneSessionClusterEntrypoint" start_jm_cmd "8081"
-
+    local c
     for (( c=0; c<${JM_KILLS}; c++ )); do
         # kill the JM and wait for watchdog to
         # create a new one which will take over
         kill_single 'StandaloneSessionClusterEntrypoint'
         wait_job_running ${JOB_ID}
     done
 
-    cancel_job ${JOB_ID}
+    for (( c=0; c<${TM_KILLS}; c++ )); do
+        sleep $(( ( RANDOM % 10 )  + 1 ))
+        kill_and_replace_random_task_manager
+        wait_job_running ${JOB_ID}
```

Comment:
`wait_job_running` can be omitted. In fact it only asserts that the job appears in the `flink list`